### DISCUSSION TITLE
Incorret background size for virtual text

### DISCUSSION THREAD

**Discussion by vdualb** [helix-editor/helix#11320] [General] [answered]
### Summary

In some cases virtual text bg is smaller than virtual text. It's confusing because in themes which have virtual text bg you expect that, when moving cursor from left to right, cursor will jump to the next symbol after right side of virtual text, but in some cases cursor appears one symbol away.

### Reproduction Steps

 

aciinema recording:
[demo.txt](https://github.com/user-attachments/files/16359086/demo.txt)

Source code:
```
fn a(smth: u32) -> f64 {
    return 30.;
}

fn main() {
    let b = a(2);

    let e = || 2;

    println!("Hello, world!");
}
```

I tried this:

1. `hx src/main.rs`
2. Go to 6th line
3. Press Right Arrow button repeatedly and notice cursor movement

I expected this to happen:
bg for virtual text on 6th line was 1 symbol longer to the right: `smth: `, not `smth:`

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Linux

### Terminal Emulator

ptyxis

### Installation Method

f40 package

### Helix Version

helix 24.3 (2cadec0b)

**Comment by pascalkuthe** [comment, marked as answer]
virtual text is not setup by us, we display whnatever the LSP sends. In this case I know for a fact that RA intentionally insert this space so that the next identifier isn't right next to the virtual text.

To display this padding the padding parameter is used. The LSP standard explicitly says that this padding should have the same background color as the main text so we are just follwoing the LSP standard here. This behaviour also matches other editor like vscode (see screenshot, colors are a bit subtle but you can see that the space was inserted for the inlay hint that has the normal background color and not the background color of the virtual hint).

![image](https://github.com/user-attachments/assets/0e039f3b-051c-4ef6-98c5-fbe9ba08ed0d)

It's an intentional choice of the language server that the text looks like this (if they wanted a separate background color they would have included a space in the text' instead of using pading) and we are respecting that in helix.

**Reply by vdualb** [reply]
Oh, that's unfortunate. I find cursor jumping on virtual text confusing and this intentional padding makes it worse. Thanks for the answer
